### PR TITLE
feat: add strict version to keys where the return type is a union of the input

### PR DESCRIPTION
**Pull request by TkDodo**
---

Make sure that you:

- [x] Typedoc added for new methods and updated for changed
- [x] Tests added for new methods and updated for changed
- [x] New methods added to `src/index.ts`
- [x] New methods added to `mapping.md`

### CODE REVIEW

File: docs/src/FnDoc.tsx [hunk 1]
```diff
@@ -38,6 +39,11 @@ export function FnDoc(props: FnDocProps) {
               <div className=" badge pipeable-color">pipeable</div>
             </small>
           )}
```
[the remark this answers is not in the document]
Comment (TkDodo) [reply docs/src/FnDoc.tsx]: done

File: src/keys.test.ts [hunk 2]
```diff
@@ -8,4 +9,13 @@ describe('Test for keys', () => {
   it('should return keys of object', () => {
     expect(keys({ a: 'x', b: 'y', c: 'z' })).toEqual(['a', 'b', 'c']);
   });
+
+  describe('strict', () => {
+    const actual = keys.strict({ a: 'x', b: 'y', c: 'z' } as const);
+    expect(actual).toEqual(['a', 'b', 'c']);
+
+    const result: AssertEqual<typeof actual, Array<'a' | 'b' | 'c'>> = true;
```
Comment (vlad-yakovlev) [inline src/keys.test.ts:17]: Have never seen this type comparison before. TS would broke here on Jest compile?

Comment (TkDodo) [reply src/keys.test.ts:17]: it's a type level "test". the jest assertion itself doesn't really do anything (compares true with true), but the typescript compiler will complain if the two types are not the same.

Comment (vlad-yakovlev) [reply src/keys.test.ts:17]: Look like possible replacement but we'll be required configure eslint for that. Think we can got back to this later, I'll create issue

**Pull request merged**